### DISCUSSION TITLE
trunk rebuilds everything when anything changes

### DISCUSSION THREAD

**Discussion by tshepang** [sycamore-rs/sycamore#164] [Q&A] [answered]
I don't know if this is normal behavior, or maybe there's some setting I'm missing.

My setup is just copied from the "hello world", only I added an empty index.html in project repo, to please trunk.

**Comment by lukechu10** [comment, marked as answer]
By "anything", do you mean non rust files e.g. css?
If that's the case, there is already an issue open on the trunk repo here: https://github.com/thedodd/trunk/issues/185

**Reply by tshepang** [reply]
it happens with a main.rs change ~as well~
```
Jul 10 21:05:47.969  INFO change detected in "/home/tshepang/bag/web/src/main.rs"
Jul 10 21:05:47.970  INFO 📦 starting build
Jul 10 21:05:47.977  INFO spawning asset pipelines
Jul 10 21:05:48.685  INFO building web
    Blocking waiting for file lock on build directory
   Compiling proc-macro2 v1.0.27
   Compiling unicode-xid v0.2.2
   Compiling autocfg v1.0.1
   Compiling wasm-bindgen-shared v0.2.74
   Compiling cfg-if v1.0.0
   Compiling bumpalo v3.7.0
   Compiling log v0.4.14
   Compiling num-traits v0.2.14
   Compiling num-integer v0.1.44
   Compiling indexmap v1.7.0
   Compiling quote v1.0.9
   Compiling syn v1.0.73
   Compiling wasm-bindgen-backend v0.2.74
   Compiling wasm-bindgen-macro-support v0.2.74
   Compiling sycamore-macro v0.5.1 (/home/tshepang/sycamore/packages/sycamore-macro)
   Compiling wasm-bindgen-macro v0.2.74
   Compiling wasm-bindgen v0.2.74
   Compiling js-sys v0.3.51
...

**Reply by lukechu10** [reply]
Do you mean it rebuilds all the dependencies as well? Trunk should only do that once. In fact, it just calls `cargo build` under the hood. I have never seen this problem.

**Reply by tshepang** [reply]
yeah, and am actually not experiencing the other bug you linked... it happens only when I change main.rs

**Reply by lukechu10** [reply]
Does it happen with a regular cargo project?

**Reply by tshepang** [reply]
it doesn't (and was not aware trunk could be used like that)

**Reply by tshepang** [reply]
the project was part of a workspace, and when I took it out, things work fine... is that a trunk bug

**Reply by lukechu10** [reply]
Right now, trunk doesn't detect changes in other crates of the same workspace. Is there some public repo that I could look at?

**Reply by tshepang** [reply]
it's not public, but I tried re-creating the problem in a new workspace, and I failed
